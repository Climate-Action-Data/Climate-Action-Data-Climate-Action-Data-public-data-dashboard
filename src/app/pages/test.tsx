'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test