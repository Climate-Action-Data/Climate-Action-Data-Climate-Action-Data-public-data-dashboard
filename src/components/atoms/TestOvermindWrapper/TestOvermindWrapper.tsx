import { createOvermind } from 'overmind'
import { Provider } from 'overmind-react'
import { config } from '@/overmind'
import { PropsWithChildren } from 'react'

const overmind = createOvermind(config)

export const TestOvermindWrapper = (props: PropsWithChildren) => {
  return <Provider value={overmind}>{props.children}</Provider>
}
