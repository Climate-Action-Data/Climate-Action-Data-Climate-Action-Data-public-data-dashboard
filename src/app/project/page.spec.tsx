import { render } from '@testing-library/react'
import PageDetails from './page'
import { TestOvermindWrapper } from '@/test/TestOvermindWrapper'

const mockPush = jest.fn()
jest.mock(`next/navigation`, () => ({
  ...jest.requireActual(`next/navigation`),
  useRouter: () => ({ push: mockPush, query: { query: `id` } }),
  useSearchParams: () => ({ get: () => `id` }),
}))

it(`renders correctly`, () => {
  const { container } = render(
    <TestOvermindWrapper>
      <PageDetails />
    </TestOvermindWrapper>,
  )
  expect(container).toMatchSnapshot()
})
