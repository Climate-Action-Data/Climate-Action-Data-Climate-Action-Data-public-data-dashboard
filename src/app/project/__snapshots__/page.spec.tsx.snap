// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-10k5p03"
      id="projectDetails"
    >
      <div
        class="css-0"
      >
        <div
          class="chakra-stack css-qlxuh7"
        >
          <svg
            class="chakra-icon css-g7llvd"
            focusable="false"
            viewBox="0 0 40 16"
          >
            <circle
              cx="1.42857"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="1.42857"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="13.8094"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="13.8094"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="26.1903"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="26.1903"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="38.5716"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="38.5716"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
          </svg>
          <h2
            class="chakra-heading css-16o2lp8"
          >
            Project Details
          </h2>
        </div>
        <div
          class="css-1mliti3"
        >
          <div
            class="chakra-container css-tqikoc"
          >
            <div
              class="chakra-stack css-j7qwjs"
            >
              <div
                class="css-s8hfc8"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Registry
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Methodology
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Project Developer
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Project Link
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Sector
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Project Type
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Status
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Status Updated On
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack__divider css-1g2u1yc"
              />
              <div
                class="css-k008qs"
              >
                <div
                  class="css-s8hfc8"
                >
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Total Available Units
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Total Issuances
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Crediting Period
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="css-s8hfc8"
                >
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Total Retirements
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Annual Estimated Units
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-uwkl65"
                    >
                      Unit Metric Measurement
                    </p>
                    <div
                      class="css-0"
                    >
                      <div
                        class="chakra-skeleton css-1tjwkx3"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack__divider css-1g2u1yc"
              />
              <div
                class="css-s8hfc8"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    NDC Coverage
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Correspondingly Adjusted
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack__divider css-1g2u1yc"
              />
              <div
                class="css-s8hfc8"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Project Tags
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Co-Benefits
                  </p>
                  <div
                    class="css-0"
                  >
                    <div
                      class="chakra-skeleton css-1tjwkx3"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="chakra-container css-11scci"
          >
            <div
              class="chakra-skeleton css-l30rjy"
            />
            <div
              class="chakra-container css-bybdqv"
            >
              <div
                class="css-s8hfc8"
              >
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Country
                  </p>
                  <p
                    class="chakra-text css-0"
                  />
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    In-country Region
                  </p>
                  <p
                    class="chakra-text css-0"
                  />
                </div>
                <div
                  class="css-0"
                >
                  <p
                    class="chakra-text css-uwkl65"
                  >
                    Geographic identifier
                  </p>
                  <p
                    class="chakra-text css-0"
                  >
                    --
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-8dnrdk"
        id="issuancesRetirements"
      >
        <div
          class="chakra-stack css-qlxuh7"
        >
          <svg
            class="chakra-icon css-g7llvd"
            focusable="false"
            viewBox="0 0 40 16"
          >
            <circle
              cx="1.42857"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="1.42857"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="13.8094"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="13.8094"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="26.1903"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="26.1903"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="38.5716"
              cy="1.42857"
              fill="currentFill"
              r="1.42857"
            />
            <circle
              cx="38.5716"
              cy="13.8097"
              fill="currentFill"
              r="1.42857"
            />
          </svg>
          <h2
            class="chakra-heading css-16o2lp8"
          >
            Issuances & Retirements
          </h2>
        </div>
      </div>
    </div>
  </div>
</div>
`;
